src/core/datastore/sync/sync-manager.js

```diff
@@ -347,4 +358,60 @@ export class SyncManager {
         return Promise.reject(new SyncError('Invalid sync event name'));
     }
   }
+
+  _getInternalPullQuery(userQuery = {}, totalCount) {
+    const { filter, sort, fields, skip } = userQuery;
+    const query = new Query({ filter, sort, fields, skip });
+    query.limit = totalCount;
+    const noSortProvided = !sort || isEmpty(sort);
+    const noSkipLimitProvided = !userQuery.skip && !userQuery.limit;
+    if (noSortProvided || noSkipLimitProvided) {
```

Comment:
Shouldn't this be `&&`? The way it reads - if a user provided a sort, but not a skip/limit, we will replace their sort with ours.

Reply:
This doesn't make sense to me. I read the doc but I am still confused here.

Reply:
We can discuss it at standup. It's the second bullet point of the section about sort.